ft_sourceparcellate.m

```diff
@@ -94,19 +94,25 @@
 end
 
 % ensure it is a parcellation, not a segmentation
-parcellation = ft_checkdata(parcellation, 'datatype', 'parcellation', 'parcellationstyle', 'indexed');
+parcellation = ft_checkdata(parcellation, 'datatype', 'parcellation', 'parcellationstyle', 'indexed', 'hasunit', 'yes');
 
 % keep the transformation matrix
 if ~isempty(transform)
   parcellation.transform = transform;
 end
 
 % ensure it is a source, not a volume
-source = ft_checkdata(source, 'datatype', 'source', 'inside', 'logical');
+source = ft_checkdata(source, 'datatype', 'source', 'insidestyle', 'logical', 'hasunit', 'yes');
 
 % ensure that the source and the parcellation are anatomically consistent
-if ~isalmostequal(source.pos, parcellation.pos, 'abstol', 1000000*eps)
-  ft_error('the source positions are not consistent with the parcellation, please use FT_SOURCEINTERPOLATE');
+if ~strcmp(source.unit, parcellation.unit)
+  ft_error('the units of the source and parcellation structure are not consistent, please use FT_SOURCEINTERPOLATE');
+end
+
+% ensure that the source and the parcellation are anatomically consistent
+tolerance = 0.1 * ft_scalingfactor('mm', source.unit);
+if ~isalmostequal(source.pos, parcellation.pos, 'abstol', tolerance)
```

Comment:
perhaps quickly check whether this is similarly handled in ft_virtualchannel. It could be that you already did this, but I am going through the functions in alphabetical order.

Reply:
good idea, I will update it there as well.